File: social-service/src/main/java/com/eirsteir/coffeewithme/social/repository/UniversityRepository.java
package com.eirsteir.coffeewithme.social.repository;

import com.eirsteir.coffeewithme.social.domain.university.University;
import com.eirsteir.coffeewithme.social.dto.NoCampusUniversity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;

import java.util.List;

public interface UniversityRepository extends JpaRepository<University, Long> {

    @Query("SELECT university.id as id, university.name as name FROM University university")
    List<NoCampusUniversity> findAllExcludeCampuses();

}

File: social-service/src/main/kotlin/com/eirsteir/coffeewithme/social/repository/UniversityRepository.kt
package com.eirsteir.coffeewithme.social.repository

import com.eirsteir.coffeewithme.social.domain.university.University
import com.eirsteir.coffeewithme.social.dto.NoCampusUniversity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query

interface UniversityRepository : JpaRepository<University, Long> {

    @Query("SELECT university.id as id, university.name as name FROM University university")
    fun findAllExcludeCampuses(): List<NoCampusUniversity>

}